### PR TITLE
test: add `no_zk` and `verify` flag for circom benchmarking

### PR DESCRIPTION
**Pull request by TomTaehoonKim**
# Description

This PR adds a `no_zk` and `verify` flag for circom benchmarking.

With the `no_zk` flag, you can create proof without zk. By default zk is enabled. You can use this flag in case you want to compare the proof with rapidsnark.
For benchmarking proving speed, we do not need to verify proof everytime. So, `verify` flag is added in case you want to verify the proof with public inputs.

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Comment by chokobole** [comment]
- [feat(zk/r1cs): add default constructor for groth16 proof](https://github.com/kroma-network/tachyon/pull/423/commits/4f8d63e73d5c519e9032e6aa9450d06775c6c0fd) Please change scope `zk/r1cs` to `zk`
- [test: add no_zk flag for circom benchmark](https://github.com/kroma-network/tachyon/pull/423/commits/ae81518b6307d5967a4a66b137a7724d6a061407) Please add scope `circom`.

File: vendors/circom/prover_main.cc [hunk 1]
```diff
@@ -75,13 +75,22 @@ void CreateProof(const base::FilePath& zkey_path,
...
+        full_assignments.subspan(constraint_matrices.num_instance_variables),
+        full_assignments.subspan(1));
+  } else {
+    proof = zk::r1cs::groth16::CreateProofWithAssignmentNoZK(
+        proving_key, absl::MakeConstSpan(h_evals),
+        full_assignments.subspan(
+            1, constraint_matrices.num_instance_variables - 1),
+        full_assignments.subspan(constraint_matrices.num_instance_variables),
+        full_assignments.subspan(1));
+  }
```
Comment (chokobole) [inline vendors/circom/prover_main.cc:93]: negation of boolean expression is hard to read. Please change as follows!

```suggestion
  if (no_zk) {
    proof = zk::r1cs::groth16::CreateProofWithAssignmentNoZK(
        proving_key, absl::MakeConstSpan(h_evals),
        full_assignments.subspan(
            1, constraint_matrices.num_instance_variables - 1),
        full_assignments.subspan(constraint_matrices.num_instance_variables),
        full_assignments.subspan(1));
  } else {
    proof = zk::r1cs::groth16::CreateProofWithAssignmentZK(
        proving_key, absl::MakeConstSpan(h_evals),
        full_assignments.subspan(
            1, constraint_matrices.num_instance_variables - 1),
        full_assignments.subspan(constraint_matrices.num_instance_variables),
        full_assignments.subspan(1));
  }
```

**Comment by chokobole** [comment]
Could you please add `verify` flag, too? I believe there's no need to verify every time!

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

File: vendors/circom/prover_main.cc [hunk 2]
```diff
@@ -130,6 +133,11 @@ int RealMain(int argc, char** argv) {
   parser.AddFlag<base::BoolFlag>(&no_zk).set_long_name("--no_zk").set_help(
       "Create proof without zk. By default zk is enabled. Use this flag to "
       "compare the proof with rapidsnark.");
+  parser.AddFlag<base::BoolFlag>(&verify)
+      .set_long_name("--verify")
```
Comment (chokobole) [inline vendors/circom/prover_main.cc:137]: Please add the description to `vendors/circom/README.md`

**Review by chokobole — APPROVED** [review]
LGTM

**Pull request merged**